smbios/fuzz.go

```diff
@@ -0,0 +1,35 @@
+// Copyright 2017 DigitalOcean.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//+build gofuzz
+
+package smbios
+
+import (
+	"bytes"
+)
+
+func Fuzz(data []byte) int {
+	return fuzzDecoder(data)
+}
+
+func fuzzDecoder(data []byte) int {
+	d := NewDecoder(bytes.NewReader(data))
+
+	if _, err := d.Decode(); err != nil {
+		return 0
```

Comment:
I'm not very familiar with `go-fuzz`, but it looks like the underlying error is getting suppressed here.  Do we really want that?  Examples from the `go-fuzz` [readme](https://github.com/dvyukov/go-fuzz), use a `panic` on errors/unexpected behavior...

Reply:
RIght, so in this case, if the error is not nil, we inform go-fuzz that the input it gave us was not "interesting".  It generates all kinds of garbage data but we want to guide it toward generating data that can be parsed as valid DMI data.

Since this package is just doing decoding, I'd expect any major problems to appear as a panic within the package itself.